### DISCUSSION TITLE
Does use of flow measurements actually solve the autocorrelation problem?

### DISCUSSION THREAD

**Discussion by tomfid** [Data4DM/BayesSD#127] [people; relating]
Problem: there's likely to be autocorrelation in the residuals for stock measurements, if the model doesn't filter to adjust for driving noise.

Proposed solution: use flow measurements instead (production starts and finishes rather than WIP inventory, for example).

Question: (to what extent) does this actually work?

In an open loop system, it should work, because the flow is a first difference and therefore whiter than the stock. But in the closed loop version, flows = f(stocks), so it seems like there should still be autocorrelated disturbances in the flows arising from the noise previously integrated into the stocks.

Is some of the real-world advantage here that the flow measurements will capture some of the driving noise influencing the stocks?

**Comment by hyunjimoon** [comment]
Isn't all argument related to prior/posterior/stock/flow relative? https://github.com/Data4DM/BayesSD/discussions/103

We can't get free from autocorrelation (we are injecting process noise, after all), but stocked flow is lesser evil (as it is one step before the stock..?) was my understanding. From the diagram below I abstracted the inventory model to trace process noise's effect on estimation, so using `stocked pping, ping` as observed data _may_ have lower autocorrelation than using `SS, S`.

<img width="880" alt="image" src="https://user-images.githubusercontent.com/30194633/207950863-af644915-e4b3-4d9e-8190-1f5f6be8e463.png">

**Reply by hyunjimoon** [reply]
> flows = f(stocks)

I agree; everything step of all compartments can be expressed with parameters and the initial stock value was Jair and my agreement. @jandraor, may I also invite you to this discussion? What do you think is a good way to explore process noise's effect on estimation? One hypothesis was better parameter recovery but worse (tighter credible interval) when using stock (not stocked flow) as observed data. (this needs detailed def. of parameter recovery though)

**Reply by tomfid** [reply]
I've been thinking about this toy version of the inventory model:
![image](https://user-images.githubusercontent.com/2151849/207961719-dd7b2e12-e251-48d4-872b-f33d6bcc4efb.png)

I think the stock/flow measurement question is mixed up with other issues about structure, particularly where noise enters the system.

- If you have measurements for producing and shipping, you might think inventory is also known ... but this isn't the case if there's unobserved stock shrinkage.
- Inventory is autocorrelated, integrating noise from producing and customer orders (via shipping), but producing is also autocorrelated, partly by correlated driving noise, and partly by its dependence on inventory and expected loss stocks.
- Measurements of producing include measurement error, but they also capture the process noise entering the true production as well. That makes part of the driving noise affecting inventory observable.
- etc.

Sorting out these effects would be pretty interesting.

**Reply by hyunjimoon** [reply]
! Interesting `corr time` is shared between order rate, producing, shrinking from your model. While building one layer deeper version of the above, I thought of using the same value (prior) for two fraction pairs: `adj info frac1` = `adj mat push frac1` and `adj info frac2` = `adj mat push frac2`. This decision is not based on managerial knowledge, but on geometric symmetry.. Would you agree, a wise 👴manager would match the two information and material delays?

<img width="1149" alt="image" src="https://user-images.githubusercontent.com/30194633/207969841-7c2eeb89-2a99-4cd6-8e8f-910f009bcfa2.png">

**Reply by tomfid** [reply]
I just made corr time a common feature for convenience - probably no reason to think that it's similar in general. I realized after I finished it that I actually should make it a Wiener process, with var~=TIME STEP, so that the integrated noise has the specified variance on the unit time scale. I think that's what the Kalman filter is assuming. If the noise itself is autocorrelated, then there really should be an explicit stock for that (though I guess there is one inside the PINK NOISE function).

**Reply by tomfid** [reply]
What is EEOR here? Looks like the expectation process is 2nd order? Is this for extrapolation, or something else?

I'm not sure the expectation adjustment rates need to match the stock adjustment rates - seems like the relationship might vary depending on what demand noise looks like.

**Reply by hyunjimoon** [reply]
My subconsciousness analysis report:
- I thought taking expectations over different adjustment times would have the effect of a layered low pass filter; e.g. if `adj info frac1` is 7 and `adj info frac2` is 30.5, perhaps we could capture multi-seasonality. Long and short-term trend at the very least like this [Aki's birthday casestudy](https://avehtari.github.io/casestudies/Birthdays/birthdays.html#Model_4:_long_term_smooth_+_seasonal_+_weekday_with_increasing_magnitude) because the structure of gaussian process' correlation time and process noise's autocorrelation all seem similar to me...

- EEOR is expected expected order delay. The idea of ever-growing hierarchical model motivated the above model where stock's stock and preparing the preparing can be imagined. In two layer model, WIP is Inventory's stock as production start rate prepares production rate. Deeper layer would have longer delay to reach to the delivery state which is why I connected that with the longer smoothed order rate (expected expected order rate). As such, in n^th layer model, we would inductively have hierarchical stock  $S_n$, preparing rate $p_n$_ing, expectations $E_n$OR (order rate). 

- [ARIMA statistical analysis differences the real world time series](https://en.wikipedia.org/wiki/Autoregressive_integrated_moving_average#Differencing) until white noise is excavated. I cannot justify enough yet, but I thought clear time series decomposition and frequency-based allocation to different levels of resources at hand is a must for efficient management.

**Reply by hyunjimoon** [reply]
In other words, seasonalities statisticians extract from real data may be generated from adjustment mechanisms (balance loop) with different aggregation level and time horizons, represented as both information and material adjustment fraction: E_n_OR and  S_n+p_n_ing.

Again, the aggregation level, time step, and time horizon are all resolution (dynamic and spatial aggregation and separation in https://github.com/Data4DM/BayesSD/issues/34, https://github.com/Data4DM/BayesSD/discussions/120)

**Reply by hyunjimoon** [reply]
Hierarchy in forecasting as information adjustment and preparing stocks as material adjustments collaborate to react to "the others" where "one-the other" relationships are overlayed in a recursive manner. Question of [how forecast accuracy affect supply chain slide](https://github.com/Data4DM/BayesSD/blob/master/DailyDigest/Reference/sim_stats/Kandemir22_ForecastingDmdPattern.pdf) gives us reference as below: 
- Aburto, L. and Weber, R. (2007) Improved supply chain management based on hybrid demand forecasts
- Disney, S.M. and Lambrecht, M.R. (2010) On replenishment rules, forecasting, and the bullwhip effect in supply chains. 2008th
- Forrester, J.W. (1958) ‘Industrial dynamics: A major breakthrough for decision makers’
- Hu, Y., Zhang, J. and Xu, Z. (2007) ‘Asymmetric demand information’s impact on supply chain performance and relationship under price-only contract’
- Lyneis, J.M. (2000) ‘System dynamics for market forecasting and structural analysis’, System Dynamics Review
- McCullen, P. and Towill, D. (2002) ‘Diagnosis and reduction of bullwhip in supply chains’
- Moran, F.V. and Barrar, P. (2006) ‘Supply Chain Dynamics: Structural Causes of the Bullwhip Effect’
- Ouyang, Y., Lago, A. and Daganzo, C.F. (2006) ‘Taming the Bullwhip Effect: From Traffic to Supply Chains’
- Sterman, J.D. (1989) ‘Modeling Managerial Behavior: Misperceptions of Feedback in a Dynamic Decision Making Experiment’
- Sterman, J.D. (2000) Business Dynamics: Systems Thinking and Modeling for a Complex World. McGraw-Hill Higher Education.
- Terwiesch, C. et al. (2005) ‘An Empirical Analysis of Forecast Sharing in the Semiconductor Equipment Supply Chain’, Management Science
- Torres, O.C. and Moran, F.V. (2006) The Bullwhip Effect in Supply Chains - A review of methods, components and cases

**Reply by hyunjimoon** [reply]
Could you send me the vensim model? I have questions about your model such as:
- is shrinkage observability multiplicative effect?
- is inflow of expected loss rate hidden? 
- where does driving noise enter?

**Comment by hyunjimoon** [comment]
@tomfid Would the following from your https://metasd.com/2012/05/bathtub-statistics/ one reason why you feel less welcoming for flow-based measurements?
> Derivatives amplify noise, and it’s easy for other things to go wrong, like getting the reporting lags misaligned.
